public/app/plugins/datasource/loki/components/LokiCheatSheet.tsx

```diff
@@ -1,43 +1,127 @@
-import React from 'react';
-
-const CHEAT_SHEET_ITEMS = [
-  {
-    title: 'See your logs',
-    label: 'Start by selecting a log stream from the Log labels selector.',
-  },
-  {
-    title: 'Logs from a "job"',
-    expression: '{job="default/prometheus"}',
-    label: 'Returns all log lines emitted by instances of this job.',
-  },
-  {
-    title: 'Combine stream selectors',
-    expression: '{app="cassandra",namespace="prod"}',
-    label: 'Returns all log lines from streams that have both labels.',
-  },
-  {
-    title: 'Search for text',
-    expression: '{app="cassandra"} (duration|latency)\\s*(=|is|of)\\s*[\\d\\.]+',
-    label: 'Add a regular expression after the selector to filter for.',
-  },
-];
-
-export default (props: any) => (
-  <div>
-    <h2>Loki Cheat Sheet</h2>
-    {CHEAT_SHEET_ITEMS.map(item => (
-      <div className="cheat-sheet-item" key={item.title}>
-        <div className="cheat-sheet-item__title">{item.title}</div>
-        {item.expression && (
-          <div
-            className="cheat-sheet-item__expression"
-            onClick={e => props.onClickExample({ refId: 'A', expr: item.expression })}
-          >
-            <code>{item.expression}</code>
+import React, { PureComponent } from 'react';
+import { shuffle } from 'lodash';
+import { ExploreStartPageProps, DataQuery } from '@grafana/ui';
+import LokiLanguageProvider from '../language_provider';
+
+// const CHEAT_SHEET_ITEMS = [
+//   {
+//     title: 'See your logs',
+//     label: '',
+//   },
+//   {
+//     title: 'Logs from a "job"',
+//     expression: '{job="default/prometheus"}',
+//     label: 'Returns all log lines emitted by instances of this job.',
+//   },
+//   {
+//     title: 'Combine stream selectors',
+//     expression: '{app="cassandra",namespace="prod"}',
+//     label: 'Returns all log lines from streams that have both labels.',
+//   },
+//   {
+//     title: 'Search for text',
+//     expression: '{app="cassandra"} (duration|latency)\\s*(=|is|of)\\s*[\\d\\.]+',
+//     label: 'Add a regular expression after the selector to filter for.',
+//   },
+// ];
+
+const DEFAULT_EXAMPLES = ['{job="default/prometheus"}'];
+const PREFERRED_LABELS = ['job', 'app', 'k8s_app'];
+
+export default class LokiCheatSheet extends PureComponent<ExploreStartPageProps, { userExamples: string[] }> {
+  userLabelTimer: NodeJS.Timeout;
+  state = {
+    userExamples: DEFAULT_EXAMPLES,
+  };
+
+  componentDidMount() {
+    this.scheduleUserLabelChecking();
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.userLabelTimer);
+  }
+
+  scheduleUserLabelChecking() {
+    this.userLabelTimer = setTimeout(this.checkUserLabels, 1000);
+  }
+
+  checkUserLabels = async () => {
+    // Set example from user labels
+    const provider: LokiLanguageProvider = this.props.datasource.languageProvider;
+    if (provider.started) {
```

Comment:
Nitpick: I think this could be also done by just calling `await provider.start()` as it returns resolved promise if it was already started so it should not double start: https://github.com/grafana/grafana/blob/572597d5d59dbd0f8c63cfa7748a2aa575b838f6/public/app/plugins/datasource/loki/language_provider.ts#L77

Would still need to check whether component is unmounted after `await` so not sure if it would actually make this shorter or less complex than the timeouts.

Reply:
I tried this before, but the language provider does not have `initialRange` yet. This interface needs to be cleaned up.